tests/allowed_addresses.go

```diff
@@ -172,12 +172,12 @@ func AllowedAddresses(network interfaces.LocalNetwork) {
 
 	// Test Relayer 1
 	log.Info("Testing Relayer 1: All sources -> All destinations")
-	relayerCleanup := testUtils.BuildAndRunRelayerExecutable(ctx, relayerConfigPath1)
+	relayerCleanup, readyChan := testUtils.RunRelayerExecutable(ctx, relayerConfigPath1)
 	defer relayerCleanup()
 
-	// Sleep for some time to make sure relayer has started up and subscribed.
-	log.Info("Waiting for the relayers to start up")
-	time.Sleep(10 * time.Second)
+	// Wait for relayer to start up
+	log.Info("Waiting for the relayer to start up")
+	testUtils.WaitForChannelClose(readyChan, 15*time.Second)
```

Comment:
Is there a reason not to encapsulate this `Wait` logic within `RunRelayerExecutable()`?

Reply:
We could do it, the main counterexample is if we want to start multiple instances or multiple services like we do in one place for now, this let's us parallelize it. Open to moving though

Reply:
I don't feel that strongly about it, but my vote would be to encapsulate it, since it would clean up a lot of the code, and since there's only one case where parallelization is used (though admittedly I could foresee more usages like that).